ibis/expr/operations/generic.py

```diff
@@ -275,11 +275,24 @@ class SimpleCase(Value):
     results: VarTuple[Value]
     default: Value
 
-    shape = rlz.shape_like("base")
-
-    def __init__(self, cases, results, **kwargs):
+    def __init__(self, base, cases, results, default):
         assert len(cases) == len(results)
-        super().__init__(cases=cases, results=results, **kwargs)
+
+        for case in cases:
+            if not rlz.comparable(base, case):
+                raise TypeError(
+                    f"Base expression {rlz._arg_type_error_format(base)} and "
+                    f"case {rlz._arg_type_error_format(case)} are not comparable"
+                )
+
+        if default.dtype.is_null() and results:
+            default = Cast(default, rlz.highest_precedence_dtype(results))
+        super().__init__(base=base, cases=cases, results=results, default=default)
```

Comment:
These changes are too aggressive for this PR IMO. Can you leave the original behavior? We should really really avoid throwing in additional operations in op construction, it makes debugging very difficult.

I question whether this is actually necessary. We should try to leave this out.

Reply:
I was trying to make this match the `__init__` of SearchedCase, but I think your right, I think this can get removed from both!

Reply:
fixed in #9559